src/vm/eetwain.cpp

```diff
@@ -3805,6 +3806,28 @@ bool UnwindEbpDoubleAlignFrame(
     {
         TADDR baseSP;
 
+#ifdef WIN64EXCEPTIONS
+        // Funclets' frame pointers(EBP) are always restored so they can access to main function's local variables.
+        // Therefore the value of EBP is invalid for unwinder so we should use ESP instead.
+        // If funclet frame layout is changed from CodeGen::genFuncletProlog() and genFuncletEpilog(),
+        // we need to change here accordingly.
+        if (pCodeInfo->IsFunclet())
+        {
+            baseSP = curESP + 8; // padding for 16byte stack alignment allocated in genFuncletProlog()
```

Comment:
@parjong GC cannot happen at any arbitrary point. It can only happen in interruptible regions of functions. When a thread is suspended for GC, a check if the thread is at GC safe point is made first. If it is not at a GC save point (an interruptible region of a function), then we hijack the return address of the function (replace it with an address of an asm hijack helper), resume the thread and wait until it either returns to the hijack helper, synchronizes with GC or we timeout (e.g. if the function we've hijacked didn't return for a long time since it called another one with a long running loop) and retry the suspension.
Only if it is at a GC safe point, we allow the GC stack walk happen from that place.
I belive prolog and epilog are non-interruptible.